src/rard/static/js/project.js

```diff
@@ -883,244 +589,3 @@ function runMoveAction(post_data, post_url) {
     },
   });
 }
-
-$("body").on("click", ".show-apparatus-criticus-form", function () {
-  // alert('todo: show the form beneath this')
-  $(".show-apparatus-criticus-form").show();
-  let inserting_at = $(this).data("index");
-  let $new_area = $("#new_apparatus_criticus_line_area");
-  $new_area.insertAfter($(this));
-  $("#id_new_apparatus_criticus_line_editor").find(".ql-editor").html("");
-  $("#update-apparatus-criticus-line").hide();
-  $("#submit-new-apparatus-criticus-line").show();
-  $("#submit-new-apparatus-criticus-line").attr("data-index", inserting_at);
-
-  $(".line-action").hide();
-  $new_area.show();
-});
-
-$("body").on("click", ".edit-apparatus-criticus-line", function () {
-  // alert('todo: show the form beneath this')
-  $(".edit-apparatus-criticus-line").show();
-  let $new_area = $("#new_apparatus_criticus_line_area");
-  let item_id = $(this).data("id");
-  let content_html = $(this).data("content");
-  $new_area.insertAfter($(this));
-  // quill needs to put things in <p> tags so make it easy for it otherwise we get
-  // unwanted automatic entering of newlines :/
-  $("#id_new_apparatus_criticus_line_editor")
-    .find(".ql-editor")
-    .html(`<p>${content_html}</p>`);
-  $("#submit-new-apparatus-criticus-line").hide();
-  $("#update-apparatus-criticus-line").show();
-  $("#update-apparatus-criticus-line").attr("data-id", item_id);
-
-  $(".line-action").hide();
-  $new_area.show();
-});
-
-function refreshOriginalTextApparatusCriticus() {
-  // now refresh content of editable area on the page
-  $(".rich-editor.enable-apparatus-criticus").each(function () {
-    let $editor = $(this).find(".ql-editor");
-
-    let html = $editor.html();
-
-    let csrf = document
-      .querySelector("meta[name='token']")
-      .getAttribute("content");
-    let headers = {};
-    headers["X-CSRFToken"] = csrf;
-
-    // any text within the original text editor on the same
-    // page might have become stale due to a change in the apparatus criticus
-    // so we send it to the server to re-index any app crit links within it.
-    // nb this doesn't save anything, just refreshes the text in the editor
-    // so that it appears correct to the user. It would be all handled correctly
-    // on saving anyway, but this is a cosmetic update so the user doesn't become confused!
-    $.ajax({
```

Comment:
I guess this approach is an alternative to what you were doing with the regex in app-crit-editor.js, but I think your way is better. If I understand correctly, this would update the app crit indexes at the expense of throwing away without warning any other changes the user might have made to the original text.